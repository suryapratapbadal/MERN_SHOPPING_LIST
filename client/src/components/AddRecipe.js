import React, { Component } from 'react';
import {
    Form,
    Button,
    FormGroup,
    Label,
    Input
} from 'reactstrap';

import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import {addRecipe} from '../actions/index'

class AddRecipe extends Component {
    constructor(props) {
        super(props);
        this.state = {
            recipeInput: '',
            item_id: this.props.item_id,
            loader: false
        };
    }

    onChange = event => {
        this.setState({ recipeInput : event.target.value})
    }

    onSubmit = event => {
        event.preventDefault();
        // this.setState({loader: true});
        const newRecipe = {
            item_id: this.state.item_id,
            name: this.state.recipeInput,
        };
        this.props.addRecipe(newRecipe);
    }

    render() {
        return (
            <Form inline onSubmit={this.onSubmit}>
                <FormGroup className="mb-2 mr-sm-2 mb-sm-0">
                    <Label for="recipe" className="mr-sm-2">Recipe</Label>
                    <Input type="text" name="recipe" id="recipe" placeholder="Recipe name.." onChange={this.onChange}/>
                    <Button outline color='primary' style={{ marginLeft: '1rem' }} disabled={this.state.recipeInput === ''}>Add</Button>
                </FormGroup>
            </Form>
        );
    }
}

export default connect (state => {
    const loader = state.itemReducer.loader || false;
    return {
        loader
    }
}, dispatch => {
    return bindActionCreators({addRecipe: addRecipe},dispatch)
})(AddRecipe)