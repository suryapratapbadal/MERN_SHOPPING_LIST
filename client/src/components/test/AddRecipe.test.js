import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

// Component to be tested
import { AddRecipe } from '../AddRecipe';


const mockupdatefn = jest.fn();
const classes = {
    button: {},
    textField: '',
};

// Component Renders
describe('<AddRecipe />', () => {

    const wrapper = shallow(<AddRecipe updateItem={mockupdatefn} item_id={'1'} classes={classes}/>);
    const event = {
        stopPropagation: jest.fn(),
        preventDefault: jest.fn(),
        target: {
            value: 'dummy'
        }
    }
    describe('render()', () => {

        test('renders the component', () => {

            expect(toJson(wrapper)).toMatchSnapshot();

        });
    });


    describe('onChange()',()=>{
        test('handle changing text field',()=>{
            wrapper.find('TextField').props().onChange(event);
            expect(wrapper.state('name')).toEqual('dummy');
        });
    });

    describe('onSubmit()',()=>{
        test('handle add recipe',()=>{
            wrapper.find('Form').props().onSubmit(event);
            expect(wrapper.state('name')).toEqual('');
            expect(mockupdatefn).toHaveBeenCalled();

        });
    });
});