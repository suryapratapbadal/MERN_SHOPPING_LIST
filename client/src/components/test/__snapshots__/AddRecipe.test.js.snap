// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddRecipe /> render() renders the component 1`] = `
<Form
  inline={true}
  tag="form"
>
  <FormGroup
    className="mb-2 mr-sm-2 mb-sm-0"
    tag="div"
  >
    <TextField
      className=""
      id="outlined-name"
      label="Recipe Name*"
      margin="normal"
      onChange={[Function]}
      required={false}
      select={false}
      value=""
      variant="outlined"
    />
    <WithStyles(Button)
      aria-label="Add"
      className={Object {}}
      color="secondary"
      disabled={true}
      mini={true}
      onClick={[Function]}
      variant="fab"
    >
      <pure(AddIcon) />
    </WithStyles(Button)>
  </FormGroup>
</Form>
`;
